src/legacy/modules/OAuth2.jsm

```diff
@@ -132,7 +140,7 @@ OAuth2.prototype = {
           },
 
           _checkForRedirect: function(aURL) {
-            if (aURL.startsWith(this._parent.completionURI)) {
+            if (aURL.match(this.completionRE)) {
```

Comment:
Alright that was quick ! This line seems to be the issue. Missing a `._parent`. Since `this.completionRE` is undefined the `.match` call returns the truthy value of `["]`  on the first redirect. We can either add the missing `._parent` or preferably  we go with something like this to mitigate any undefined issues in the future:
```suggestion
            if (this._parent.completionRE?.test?.(aURL)) { 
```

Reply:
Thank you! I think we can go without optional chaining for completionRE. It specifically gets set further up by calling `randomizePort`, so I don't see a situation where it would not be set.